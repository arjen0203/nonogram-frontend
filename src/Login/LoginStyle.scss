@import './../colors';

.center {
    justify-content: center;
    display: flex;

    .login-fields {
        display: flex;
        flex-direction: column;
        background-color: $blackCoral;
        padding: 20px;
        border-radius: 20px;
        margin-top: 60px;
        color: $white;

        .login-title {
            text-align: center;
            color: $white;
        }
    }
}
