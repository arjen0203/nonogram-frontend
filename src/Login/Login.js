import React, {Component} from 'react';
import './LoginStyle.scss';
import {Link} from "react-router-dom";

class Login extends Component {
    render() {
        return (
            <div className="center">
                <div className="login-fields">
                    <b className="login-title">Login</b>
                    <label>Username:</label>
                    <input className="login-username-input" type="text" placeholder=" Username"></input>
                    <label>Password:</label>
                    <input className="login-username-input" type="password" placeholder=" Password"></input>
                    <button className="login-submit-button">Login</button>

                    <Link className="to-register-link" to={'/register'} >Don't have an account yet? Register here.</Link>
                </div>
            </div>
        );
    }
}

export default Login;