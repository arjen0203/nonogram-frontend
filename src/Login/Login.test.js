import React from "react";
import ReactDOM from 'react-dom';
import Login from "./Login";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const div = document.createElement('div');
	ReactDOM.render(<Login></Login>, div);
});

it('matches snapshot', function () {
	const tree = renderer.create(<Login></Login>).toJSON();
	expect(tree).toMatchSnapshot();
});