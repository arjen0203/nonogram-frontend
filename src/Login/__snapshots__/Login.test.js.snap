// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="center"
>
  <div
    className="login-fields"
  >
    <b
      className="login-title"
    >
      Login
    </b>
    <label
      htmlFor="username"
    >
      Username:
    </label>
    <input
      className="login-username-input"
      id="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value="test123"
    />
    <label
      htmlFor="password"
    >
      Password:
    </label>
    <input
      className="login-username-input"
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="test123"
    />
    <button
      className="login-submit-button"
      onClick={[Function]}
    >
      Login
    </button>
    <b
      className="login-error"
    />
    <div
      className="to-register-link"
      to="/register"
    >
      Don't have an account yet? Register here.
    </div>
  </div>
</div>
`;
